pkg/kn/commands/service/import.go

```diff
@@ -0,0 +1,143 @@
+)
+
+// NewServiceImportCommand returns a new command for importing a service.
+func NewServiceImportCommand(p *commands.KnParams) *cobra.Command {
+	var waitFlags commands.WaitFlags
+
+	command := &cobra.Command{
+		Use:   "import FILENAME",
+		Short: "Import a service and its revisions",
+		Example: `
+ # Import a service in YAML format
+ kn service import /path/to/file.yaml
+
+ # Import a service in JSON format
+ kn service import /path/to/file.json`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				return errors.New("'kn service import' requires filename of import file as single argument")
+			}
+			filename := args[0]
+
+			namespace, err := p.GetNamespace(cmd)
+			if err != nil {
+				return err
+			}
+
+			client, err := p.NewServingClient(namespace)
+			if err != nil {
+				return err
+			}
+
+			return importWithOwnerRef(client, filename, cmd.OutOrStdout(), waitFlags)
+		},
+	}
+	flags := command.Flags()
+	commands.AddNamespaceFlags(flags, false)
+	waitFlags.AddConditionWaitFlags(command, commands.WaitDefaultTimeout, "import", "service", "ready")
+
+	return command
+}
+
+func importWithOwnerRef(client clientservingv1.KnServingClient, filename string, out io.Writer, waitFlags commands.WaitFlags) error {
+	var export clientv1alpha1.Export
+	file, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+	decoder := yaml.NewYAMLOrJSONDecoder(file, 512)
+	err = decoder.Decode(&export)
+	if err != nil {
+		return err
+	}
+	if export.Spec.Service.Name == "" {
+		return fmt.Errorf("provided import doesn't content service name, please note that only kn' custom Export format is supported")
+	}
+
+	serviceName := export.Spec.Service.Name
+
+	// Return error if service already exists
+	svcExists, err := serviceExists(client, serviceName)
+	if err != nil {
+		return err
+	}
+	if svcExists {
+		return fmt.Errorf("cannot import service '%s' in namespace '%s' because the service already exists",
+			serviceName, client.Namespace())
+	}
+
+	err = client.CreateService(&export.Spec.Service)
+	if err != nil {
+		return err
+	}
+
+	// Retrieve current Configuration to be use in OwnerReference
+	retries := 0
+	var currentConf *servingv1.Configuration
+	for {
+		currentConf, err = client.GetConfiguration(serviceName)
+		if err != nil {
+			if apierrors.IsNotFound(err) && retries < 5 {
+				retries++
+				time.Sleep(time.Second)
+				continue
+			} else {
+				return err
+			}
+		}
+		break
+	}
+
+	// Create revision with current Configuration's OwnerReference
+	if len(export.Spec.Revisions) > 0 {
+		for _, r := range export.Spec.Revisions {
+			tmp := r
+			// OwnerRef ensures that Revisions are recognized by controller
+			tmp.OwnerReferences = []metav1.OwnerReference{*kmeta.NewControllerRef(currentConf)}
+			if err = client.CreateRevision(&tmp); err != nil {
+				return err
+			}
+		}
+	}
+
+	err = waitIfRequested(client, serviceName, waitFlags, "Importing", "imported", out)
```

Comment:
a service has 5 revisions among which 2 are active (configured in traffic block); will service still report true if any of those 3 non-active revisions failed to become ready ?

Reply:
The current `service export` impl doesn't produce such export output, only active revisions are [included](https://github.com/knative/client/blob/master/pkg/kn/commands/service/export.go#L245).

Reply:
I wonder whether we should allow later in `service export`, too, to export all revisions, so that the import then also would import the history. But let's leave that question for future refinement of this feature.